tests/unit/functions.tcl

```diff
@@ -1078,4 +1078,21 @@ start_server {tags {"scripting"}} {
         r config set replica-serve-stale-data yes
         set _ {}
     } {} {external:skip}
+
+    test {FUNCTION - redis version api} {
+        r FUNCTION load lua test replace { 
+            local version = redis.REDIS_VERSION_NUM
+
+            redis.register_function{function_name='get_version_v1', callback=function()
+              return string.format('%s.%s.%s',
+                                    bit.band(bit.rshift(version, 4), 0x000000ff),
+                                    bit.band(bit.rshift(version, 2), 0x000000ff),
+                                    bit.band(version, 0x000000ff))
+            end}
+            redis.register_function{function_name='get_version_v2', callback=function() return redis.REDIS_VERSION end}
+        }
+
+        catch {[r fcall f1 0]} e
+        assert_equal  [r fcall get_version_v1 0] [r fcall get_version_v2 0]
+    }
```

Comment:
just wanna be sure i'm not missing something.
with such big functions, (.e.g the multi-line one that does `string.format`), we normally expect people to declare the function procedure first, and then just use it's name in the registration line, right?

p.s. i see one variable is used (accessed) in the registration time code, and the other in the function (runtime) code itself, i suppose that's on purpose (to test that it's accessible on both stages), right?

Reply:
> just wanna be sure i'm not missing something.
with such big functions, (.e.g the multi-line one that does string.format), we normally expect people to declare the function procedure first, and then just use it's name in the registration line, right?

Users can do both and it up to the user decide what he prefer, for us it does not matter, both will work.

> p.s. i see one variable is used (accessed) in the registration time code, and the other in the function (runtime) code itself, i suppose that's on purpose (to test that it's accessible on both stages), right?

Yes, I want to make sure (in the test) that the API is give on function loading and on function invocation.